core/server/services/bulk-email/index.js

```diff
@@ -46,24 +47,45 @@ module.exports = {
      * @param {Email} message - The message to send
      * @param {[EmailAddress]} recipients - the recipients to send the email to
      * @param {[object]} recipientData - list of data keyed by email to inject into the email
-     * @returns {Promise<boolean>} A promise representing the success of the email sending
+     * @returns {Array{Promise<boolean>}} An array of promises representing the success of the batch email sending
      */
     async send(message, recipients, recipientData) {
+        let BATCH_SIZE = 1000;
+
         if (!mailgunInstance) {
             return;
         }
         let fromAddress = message.from;
-        if (/@localhost$/.test(message.from)) {
+        if (/@localhost$/.test(message.from) || /@ghost.local$/.test(message.from)) {
             fromAddress = 'localhost@example.com';
             common.logging.warn(`Rewriting bulk email from address ${message.from} to ${fromAddress}`);
+
+            BATCH_SIZE = 2;
         }
         try {
-            const messageData = Object.assign({}, message, {
-                to: recipients.join(', '),
-                from: fromAddress,
-                'recipient-variables': recipientData
+            let chunks = _.chunk(recipients, BATCH_SIZE);
+
+            return Promise.map(chunks, (chunk) => {
+                const messageData = Object.assign({}, message, {
+                    to: chunk.join(', '),
+                    from: fromAddress,
+                    'recipient-variables': recipientData
```

Comment:
@gargol I think this will need adjusting to match the chunk too, otherwise we're sending recipient data for all email addresses for each message.

Reply:
true that! :fishing_pole_and_fish: